### PR TITLE
[IMP] mail: move discuss component to the correct bundle

### PR DESCRIPTION
**Pull request by tsm-odoo**
The mail module is split into several sub-folders to determine when the files should be loaded (e.g., web, public, public_web...).

Currently, several code snippets are misplaced, which increases the load on public pages or live chat.

Moreover, this can lead to dependency hell, especially when moving a feature to another bundle.

This PR moves the discuss component, which are only used in the web and public bundle, to the public_web bundle.

Part of task-3972988

### CODE REVIEW

File: addons/mail/static/src/core/web/discuss_client_action.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Discuss } from "@mail/core/common/discuss";
+import { Discuss } from "@mail/core/public_web/discuss";
```
Comment (seb-odoo) [inline addons/mail/static/src/core/web/discuss_client_action.js:1]: client action is web client only, there is no action manager on public page

File: addons/mail/__manifest__.py [hunk 2]
```diff
@@ -225,8 +227,10 @@
 
             'mail/static/src/model/**/*',
             'mail/static/src/core/common/**/*',
+            'mail/static/src/core/public_web/**/*',
             'mail/static/src/**/common/**/*',
             'mail/static/src/**/public/**/*',
+            'mail/static/src/**/public_web/**/*',
```
Comment (seb-odoo) [inline addons/mail/__manifest__.py:233]: public_web should go before public (it's more generic)

File: addons/mail/static/src/core/web/discuss_client_action.js (hunk 1 above)
Comment (tsm-odoo) [reply addons/mail/static/src/core/web/discuss_client_action.js:1]: I plan on removing `discuss_public` and renaming the client action component. We don't really care it is an action, its role is only to restore discuss thread whether it comes from an action or else.

Comment (tsm-odoo) [reply addons/mail/static/src/core/web/discuss_client_action.js:1]: Also even if we were to keep `discuss_client_action` in web, this diff is correct: discuss is effectively needed in both web and public and available in the web scope.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**